### PR TITLE
dist: adapt develhelp in Makefile 

### PR DESCRIPTION
**Pull request by smlng**
factored out from #8029, remove obsolete comment on develhelp.

### CODE REVIEW

File: dist/Makefile [hunk 1]
```diff
@@ -22,11 +22,6 @@ RIOTBASE ?= $(CURDIR)/../../RIOT
 # with the target all-valgrind instead of all:
 # make -B clean all-valgrind
 
-# Uncomment this to enable code in RIOT that does safety checking
```
Comment (kaspar030) [inline dist/Makefile]: IMO this was useful and should be adapted to use the new DEVELHELP variable.

Comment (smlng) [reply dist/Makefile]: I agree, but with the _new_ switch, `DEVELHELP` can enabled via env variable from everywhere, so no need to have this `comment out` option here. But I could leave a comment here, how to enable DEVELHELP, instead?

Comment (kaspar030) [reply dist/Makefile]: I'd keep it as before, but no using ```#DEVELHELP = 1```

**Comment by smlng** [comment]
rebased

**Comment by smlng** [comment]
ping @kaspar030

**Comment by kaspar030** [comment]
ACK&go.

**Pull request merged**